include/logging/log_core.h

```diff
@@ -802,7 +802,7 @@ __syscall void z_log_hexdump_from_user(uint32_t src_level_val,
 		break; \
 	} \
 	if (IS_ENABLED(CONFIG_LOG2)) { \
-		z_log_msg2_runtime_vcreate(CONFIG_LOG_DOMAIN_ID, _source, \
+		z_log_msg2_runtime_vcreate(CONFIG_LOG_DOMAIN_ID, _dsource, \
```

Comment:
you need to use static or dynamic source structure depending on runtime filtering option, like here: https://github.com/zephyrproject-rtos/zephyr/blob/e186ebc3143e642f69c9379abdccccf4f7fbfaf8/include/logging/log_core.h#L345

Reply:
fixed